### PR TITLE
global: backports Flask-IIIF extension from 2.1 to 2.0

### PR DESCRIPTION
**Pull request by blixhavn**

### CODE REVIEW

**Comment by blixhavn** [comment]
Added `Amended-by` as I didn't feel any of the other fits the purpose. Should I change it?

**Comment by tiborsimko** [comment]
This PR backports a new feature into an already-released maintenance branch.  Two general comments: (1) it would be preferred to fix any bugs you spotted in v2.1 so that you could use the latest version instead; (2) if not possible, and you'd need to backport a feature, then it should be optional and not switched on by default, so that other installations aren't affected...

**Comment by blixhavn** [comment]
1) Unfortunately, there are too many large and critical bugs in 2.1 to make it usable in production. (e.g. search with special characters not working [https://github.com/inveniosoftware/invenio/issues/3296], facets not working [https://github.com/inveniosoftware/invenio/issues/3333], the collection admin interface is lacking basic functionality)

2) As per our talk, I have removed the addition of `flask_iiif` to the extensions list to avoid adding features to existing 2.0 installations. In addition, I have now removed Flask-IIIF from pip requirements.

**Comment by jirikuncar** [comment]
> I have now removed Flask-IIIF from pip requirements.

You can add it as optional extra (e.g. `iiif`).

**Comment by jirikuncar** [comment]
> 15: M102 unrecognized bullet/signature

Please use one of our agreed signatures. We should also remove the bullets with label `NEW`.

**Comment by blixhavn** [comment]
There, `iiif` added as optional extra, and commit message fixed.

**Comment by jirikuncar** [comment]
@blixhavn can you mention sha1 of the "Original commit".

WDYT about following commit message:
```
* NOTE Backports original commit <sha1>. The extension is not enabled by default in order to avoid ...
```

**Comment by blixhavn** [comment]
Looks good. Amended.

File: setup.py [hunk 1]
```diff
@@ -199,6 +199,9 @@ def run(self):
     "github": [
         "github3.py>=0.9"
     ],
+    "iiif": [
```
Comment (jirikuncar) [inline setup.py]: Can you please put it before `img`?  We try to keep it sorted. Thanks

**Comment by jirikuncar** [comment]
After https://github.com/inveniosoftware/invenio/pull/3338#discussion_r34462813 and rebase we can :shipit:

**Comment by jirikuncar** [comment]
@blixhavn please finish sentence with a dot. 

```
* NOTE Backports original commit
  <sha1>. The extension is not ...
```

**Comment by jirikuncar** [comment]
@blixhavn `s/orig\./original/` + rebase

**Comment by blixhavn** [comment]
`original` was shortened to allow for the full sha1 within 72 characters. Should write `original` and add a new line before the sha1?

**Comment by jirikuncar** [comment]
@blixhavn see https://github.com/inveniosoftware/invenio/pull/3338#issuecomment-120932651 ;)

**Comment by blixhavn** [comment]
Which sentence is not finished with dot now?

**Comment by jirikuncar** [comment]
```
* NOTE Backports original commit
  213b6f1144734c9ecf425a1bc7b78e56ee5e4e3e.  The extension is not 
  enabled by default in order to avoid feature addition to existing
  minor release.
```

**Comment by jirikuncar** [comment]
```
* NOTE Backports Flask-IIIF extension from original commit ...
```

**Pull request merged**